src/panoptes/pocs/focuser/astromechanics.py

```diff
@@ -22,14 +22,17 @@ class Focuser(AbstractSerialFocuser):
     as they are marked with the decorator @abstractmethod, we have to override them.
     """
 
-    def __init__(self, name='Astromechanics Focuser', model='Canon EF-232', vendor_id=None,
-                 product_id=None, *args, **kwargs):
+    def __init__(self, name='Astromechanics Focuser', model='Canon EF-232', port=None,
+                 vendor_id=None, product_id=None, *args, **kwargs):
         # Check if have device, raise error.NotFound if unable to find.
-        self._serial_number = find_serial_port(vendor_id, product_id)
-        self._vendor_id = vendor_id
-        self._product_id = product_id
+        try:
+            port = find_serial_port(vendor_id, product_id)
+            self._vendor_id = vendor_id
+            self._product_id = product_id
+        except error.NotFound:
+            self.logger.debug(f'Could not find device port for {vendor_id=} and {product_id=}')
```

Comment:
The focuser has no `logger` attribute since the parent class hasn't been initialised yet.